client/state/utils.js

```diff
@@ -91,6 +93,14 @@ export const keyedReducer = ( keyName, reducer ) => {
 	const initialState = reducer( undefined, { type: '@@calypso/INIT' } );
 
 	return ( state = {}, action ) => {
+		if ( DESERIALIZE === action.type ) {
+			return deserializer( state );
+		}
+
+		if ( SERIALIZE === action.type ) {
+			return serializer( mapValues( state, item => reducer( item, action ) ) );
```

Comment:
yeah I thought about this but didn't know if it was right to try and optimize away. we could leave it all up to the serializer and deserializer to decide if we should map the action across the collection